### PR TITLE
RE-187 Resolve newton build/test failures

### PR DESCRIPTION
**Pull request by odyssey4me**
This PR pulls together all the fixes necessary to allow newton to build and deploy again, and also includes an update to the release version and SHA's in preparation for cutting the 14.2 stabilisation branch.

Issue: [RE-113](https://rpc-openstack.atlassian.net/browse/RE-113)

Issue: [RE-185](https://rpc-openstack.atlassian.net/browse/RE-185)

Issue: [RE-187](https://rpc-openstack.atlassian.net/browse/RE-187)

Issue: [RE-152](https://rpc-openstack.atlassian.net/browse/RE-152)

Issue: [RE-189](https://rpc-openstack.atlassian.net/browse/RE-189)

Issue: [RE-184](https://rpc-openstack.atlassian.net/browse/RE-184)

Issue: [RE-155](https://rpc-openstack.atlassian.net/browse/RE-155)

Issue: [RE-202](https://rpc-openstack.atlassian.net/browse/RE-202)

Issue: [RE-204](https://rpc-openstack.atlassian.net/browse/RE-204)

Issue: [RE-206](https://rpc-openstack.atlassian.net/browse/RE-206)

Issue: [RE-213](https://rpc-openstack.atlassian.net/browse/RE-213)

Issue: [RE-214](https://rpc-openstack.atlassian.net/browse/RE-214)

Issue: [RE-215](https://rpc-openstack.atlassian.net/browse/RE-215)

Issue: [RE-218](https://rpc-openstack.atlassian.net/browse/RE-218)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_xenial_deploy_swift

**Comment by odyssey4me** [comment]
The minorupgrade and leapfrogupgrade jobs cannot pass until there are artifacts available for the new release. As such I have kicked off the artifact build pipeline. https://rpc.jenkins.cit.rackspace.net/view/Artifacts/job/RPC-Artifact-Build_newton-periodic/22/

**Comment by odyssey4me** [comment]
Newton 14.2 artifacts are being rebuilt: https://rpc.jenkins.cit.rackspace.net/blue/organizations/jenkins/RPC-Artifact-Build_newton-periodic/detail/RPC-Artifact-Build_newton-periodic/32/pipeline/

Once that's done, these jobs will need to be rechecked.

**Comment by evrardjp** [comment]
recheck_cit_trusty_leapfrogupgrade_swift
recheck_cit_trusty_deploy_swift

**Comment by evrardjp** [comment]
recheck_cit_artifact

**Comment by odyssey4me** [comment]
The trusty-deploy-swift job failed due to https://rpc-openstack.atlassian.net/browse/RE-206, whereas the other jobs failed in unrelated testing.

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_issue_link

**Comment by odyssey4me** [comment]
Artifact build failed due to https://rpc-openstack.atlassian.net/browse/RE-211 which should be fixed by https://github.com/rcbops/rpc-gating/pull/397

**Comment by evrardjp** [comment]
Gates gates gates!

**Comment by odyssey4me** [comment]
Artifact, minorupgrade and leapfrog builds all failed due to https://rpc-openstack.atlassian.net/browse/RE-211 which should be fixed by rcbops/rpc-gating#397

trusty_deploy_swift failed due to kibana selenium tests failing

**Comment by odyssey4me** [comment]
recheck_trusty_deploy_swift

**Comment by odyssey4me** [comment]
recheck_cit_trusty_deploy_swift

**Comment by odyssey4me** [comment]
recheck_cit_artifact
recheck_cit_trusty_leapfrogupgrade_swift
recheck_cit_trusty_minorupgrade_swift
recheck_cit_xenial_minorupgrade_swift

**Comment by odyssey4me** [comment]
recheck_cit_artifact pubcloud instance build failure

**Comment by odyssey4me** [comment]
recheck_cit_trusty_deploy_swift transient kibana test failure

File: scripts/functions.sh [hunk 1]
```diff
@@ -86,6 +85,30 @@ function check_submodule_status {
...
+    if [[ -e /etc/openstack_deploy/user_rpcm_variables.yml ]]; then
+      rm -f /etc/openstack_deploy/user_rpcm_variables.yml
+    fi
+    if [[ -e /etc/openstack_deploy/user_rpcm_default_variables.yml ]]; then
+      rm -f /etc/openstack_deploy/user_rpcm_default_variables.yml
+    fi
+
+    # Copy the default override files if they do not exist
+    if [[ ! -f "${OA_OVERRIDES}" ]]; then
+      cp "${RPCD_DIR}/${OA_OVERRIDES}" "${OA_OVERRIDES}"
```
Comment (evrardjp) [inline scripts/functions.sh]: nit: "/" not needed

File: scripts/functions.sh [hunk 2]
```diff
@@ -86,6 +85,30 @@ function check_submodule_status {
...
+    fi
+
+    if [[ ! -f "${RPCD_OVERRIDES}" ]]; then
+      cp "${RPCD_DIR}/${RPCD_OVERRIDES}" "${RPCD_OVERRIDES}"
```
Comment (evrardjp) [inline scripts/functions.sh]: same here.

File: scripts/functions.sh (hunk 1 above)
Comment (odyssey4me) [inline scripts/functions.sh]: Yep, I put it there for readability - not for function.

File: scripts/functions.sh (hunk 2 above)
Comment (odyssey4me) [inline scripts/functions.sh]: as above ;)

**Comment by odyssey4me** [comment]
If this passes all tests and someone decides to merge it - please do *not* squash merge it. I need to be able to pick some of these forward to master/ocata.

**Comment by odyssey4me** [comment]
recheck_cit_trusty_leapfrogupgrade_swift selenium failure after leapfrog

**Comment by evrardjp** [comment]
nooooo

**Comment by odyssey4me** [comment]
recheck_cit_trusty_leapfrogupgrade_swift selenium failure after leapfrog

**Pull request merged**